.circleci/config.yml

```diff
@@ -169,8 +169,7 @@ commands:
 parameters:
   android-docker-image:
     type: string
-    # Hash points to previous version with node 12. When everything works with node 14 it can be removed
-    default: "circleci/android:api-29-node@sha256:71d61d6c21b29948d57120f476a83cc322a280979bce355c5a0ad771293ca380"
+    default: "2023.07-node"
```

Comment:
Since we pick the needed node/npm version via `nvm` ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/blob/b630fc00af0411c493edce81cca4021bb07c6079/.circleci/config.yml#L140-L158)), we can use the latest version of CircleCI's Android docker image. Note the tag is different because the previous one was deprecated, more information can be found in [Legacy Convenience Image Deprecation](https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034).